src/Build/Instance/TaskRegistry.cs

```diff
@@ -386,7 +387,7 @@ IFileSystem fileSystem
                 taskFactoryParameters.Add(XMakeAttributes.architecture, architecture == String.Empty ? XMakeAttributes.MSBuildArchitectureValues.any : architecture);
             }
 
-            taskRegistry.RegisterTask(taskName, AssemblyLoadInfo.Create(assemblyName, assemblyFile), taskFactory, taskFactoryParameters, parameterGroupAndTaskElementRecord);
+            taskRegistry.RegisterTask(taskName, AssemblyLoadInfo.Create(assemblyName, assemblyFile), taskFactory, taskFactoryParameters, parameterGroupAndTaskElementRecord, loggingService, buildEventContext, projectUsingTaskXml, overrideUsingTask.Equals("true", StringComparison.OrdinalIgnoreCase));
```

Comment:
```suggestion
            taskRegistry.RegisterTask(taskName, AssemblyLoadInfo.Create(assemblyName, assemblyFile), taskFactory, taskFactoryParameters, parameterGroupAndTaskElementRecord, loggingService, buildEventContext, projectUsingTaskXml, ConversionUtilities.ValidBooleanTrue(overrideUsingTask));
```

(this handles more cases including some terrifying ones)

Reply:
Do we have any data on how often some of the less-used "valid booleans" are actually used? The reason I ask—if 99.9999% of people say true/false, 99.9999% of people are wasting time looking through all the other options in the "false" case—and we check truthiness or falsiness very often in MSBuild. The 0.0001% of people would have to change their code, so it's clearly a breaking change, but it would be a pretty straightforward change. Just curious if we've ever done a cost-benefit analysis.

--Troublemaker

Reply:
I'd like to see a profile showing that it's a significant cost before considering this further.

Reply:
I tested this, and it didn't seem nearly as impactful as I'd hoped. There may be an argument for reordering them, but I would count it very low priority.